dashboard/app/controllers/abuse_report_controller.rb

```diff
@@ -0,0 +1,43 @@
+require 'json'
+require 'httparty'
+
+class AbuseReportController < ApplicationController
```

Comment:
I think it's fine for this to be a separate controller.  One nit: the route and actions are all `report_abuse*` and this is `abuse_report*`.